src/role.ts

```diff
@@ -0,0 +1,380 @@
+      namespace: this.metadata.namespace!,
+      role: this,
+    });
+    binding.addSubjects(...subjects);
+    return binding;
+  }
+}
+
+/**
+ * Represents a cluster-level role.
+ */
+export interface IClusterRole extends IResource {
+
+}
+
+/**
+ * Options for `ClusterRole`.
+ */
+export interface ClusterRoleProps extends RoleCommonProps {
+  /**
+   * Specify labels that should be used to locate ClusterRoles, whose rules
+   * will be automatically filled into this ClusterRole's rules.
+   */
+  readonly aggregationLabels?: { [key: string]: string };
+}
+
+/**
+ * ClusterRole is a cluster level, logical grouping of PolicyRules that can be
+ * referenced as a unit by a RoleBinding or ClusterRoleBinding.
+ */
+export class ClusterRole extends RoleBase implements IClusterRole {
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  public readonly resourceType = 'clusterroles';
+
+  private readonly _labelSelector: Record<string, string>;
+
+  constructor(scope: Construct, id: string, props: ClusterRoleProps = {}) {
+    super(scope, id);
+
+    this._labelSelector = props.aggregationLabels ?? {};
+
+    this.apiObject = new k8s.KubeClusterRole(this, 'Resource', {
+      metadata: props.metadata,
+      rules: Lazy.any({ produce: () => this.synthesizeRules() }),
+      aggregationRule: Lazy.any({ produce: () => this.synthesizeAggregationRules() }),
+    });
+  }
+
+  /**
+   * Add the rules of the argument ClusterRole into this ClusterRole.
+   * @param role
+   */
+  public aggregateFrom(role: ClusterRole): void {
+    const name = Names.toLabelValue(this);
+    const selector = `cdk8s.cluster-role/aggregate-to-${name}`;
+    role.metadata.addLabel(selector, 'true');
+    this._labelSelector[selector] = 'true';
+  }
+
+  private synthesizeRules(): any {
+    return undefinedIfEmpty(this._rules.map(rule => rule._toKube()));
+  }
+
+  private synthesizeAggregationRules(): k8s.AggregationRule | undefined {
+    if (Object.keys(this._labelSelector).length === 0) {
+      return undefined;
+    }
+
+    return { clusterRoleSelectors: [{ matchLabels: this._labelSelector }] };
+  }
+
+  /**
+   * Create a RoleBinding that binds the permissions in this ClusterRole
+   * to a list of subjects, that will only apply to the given namespace.
+   * @param namespace the namespace to limit permissions to.
+   * @param subjects a list of subjects to bind to
+   */
+  public bindInNamespace(namespace: string, ...subjects: ISubject[]): RoleBinding {
+    const binding = new RoleBinding(this, 'RoleBinding', {
+      namespace,
+      role: this,
+    });
+    binding.addSubjects(...subjects);
+    return binding;
+  }
+
+  /**
+   * Create a ClusterRoleBinding that binds the permissions in this
+   * ClusterRole to a list of subjects, without namespace restrictions.
+   * @param subjects a list of subjects to bind to
+   */
+  public bindInCluster(...subjects: ISubject[]): ClusterRoleBinding {
```

Comment:
I think we can use `bind` here. 

```suggestion
  public bind(...subjects: ISubject[]): ClusterRoleBinding {
```

This way its consistent with the `bind` API in `Role`. So we have:

- `role.bind -> RoleBinding`
- `clusterRole.bind -> ClusterRoleBinding`
- `clusterRole.bindInNamespace` -> `RoleBinding`.